pytest_fixtures/xdist.py

```diff
@@ -1,13 +1,52 @@
 """Fixtures specific to or relating to pytest's xdist plugin"""
+import logging
+
 import pytest
+from broker.broker import VMBroker
 
+import robottelo
 from robottelo.config import settings
 
+logger = logging.getLogger('robottelo')
+
 
 @pytest.fixture(scope="session", autouse=True)
-def align_xdist_satellites(worker_id):
-    """Set a different Satellite per worker when available in robottelo's config"""
-    settings.configure()
-    settings.server.hostname = settings.server.get_hostname(worker_id)
-    settings.configure_nailgun()
+def align_to_satellite(worker_id, satellite_factory):
+    """Attempt to align a Satellite to the current xdist worker"""
+    cache_proxy = robottelo.config.settings_proxy._cache
+    # clear any hostname that may have been previously set
+    cache_proxy['server.hostname'], on_demand_sat = None, None
+    import random
+
+    if worker_id == 'master':
+        worker_pos = 0
+    else:
+        worker_pos = int(worker_id.replace('gw', ''))
```

Comment:
Can be a one liner :

```suggestion
    worker_pos = 0 if worker_id == 'master' else int(worker_id.replace('gw', ''))
```

Reply:
I'll leave as-is for readability.